routers/repo/http.go

```diff
@@ -364,14 +364,23 @@ func serviceRPC(h serviceHandler, service string) {
 	}
 
 	if h.cfg.OnSucceed != nil {
-		input, err = ioutil.ReadAll(reqBody)
+		tmpfile, err := ioutil.TempFile("", "gogs")
```

Comment:
https://github.com/golang/go/blob/master/src/io/ioutil/tempfile.go#L55

Is this what you are looking for?